### ISSUE TITLE
Latest examples do not render outputs

### ISSUE THREAD

**Issue by echarles** [jupyterlab/jupyterlab#14221]
I am running the notebook example and after https://github.com/jupyterlab/jupyterlab/commit/2142d76f917032d332645d9fb0cde3660432d522#diff-81f7333ccbe5cab9f436c43df103c8158eea12521445f33a95e58a06940a1e13 the outputs are not rendered anymore  although the  kernel message are sent 

cc @krassowski 

![Untitled6](https://user-images.githubusercontent.com/226720/226104572-68412b64-9326-4b60-80a1-1bb8a2ccc712.gif)

**Comment by echarles** [comment]
cc/ @fcollonval

**Comment by jtpio** [comment]
Also there is no syntax highlighting. Not sure if that was the case before already though, or also related to https://github.com/jupyterlab/jupyterlab/pull/13639.

**Comment by echarles** [comment]
Thx for looking at this @jtpio and confirming the issue. I have tried to roll back a few commits behind, and even a few releases alpha behind. I had issues to make it work and tried to pin the alpha (npm/yarn resolution is not very friendly with alpha handling...), but was not able to nail down the commit that make the example fail.

Having more tests for the examples will help for sure.

**Comment by akloster** [comment]
I ran into this issue as well, and it is a little annoying. I tried rolling back some versions of jupyter, but so far haven't gotten the notebook example to work.

**Comment by echarles** [comment]
> I ran into this issue as well, and it is a little annoying. I tried rolling back some versions of jupyter, but so far haven't gotten the notebook example to work.

Thx @akloster for your inputs. It is now taken in the `Milestone 4.0.0` so hopefully will be solved before the final 4.0.0. The Beta release track has just started https://github.com/jupyterlab/jupyterlab/issues/9647#issuecomment-1484726405.

I have debugged more with the beta version trying to identify the source of the issue without success. It looks to me no developer is looking at this, so pinging @jupyterlab/jupyterlab-council to raise a bit of attention and get feedback on the next steps.

**Comment by jtpio** [comment]
Generally speaking we should address issues added to the `4.0` milestone before the final release.

And move them to `4.0.x` / `4.1` if they are not critical for the release.

**Comment by echarles** [comment]
Example of [7227449db8db478dc5e31bbc1ffdfb741766a35d ](https://github.com/jupyterlab/jupyterlab/tree/7227449db8db478dc5e31bbc1ffdfb741766a35d) on Jan 25 2023 are working fine.

**Comment by echarles** [comment]
I have gone back in time and https://github.com/jupyterlab/jupyterlab/commit/2142d76f917032d332645d9fb0cde3660432d522 is confirmed to break this. I have tried to update the cell example without success.

FAILING 4947d6dd953ca6ac935d34b95bfdba2b02bcfbda Mar 16
FAILING fc773a380ce15ec812c615e0fa154b38ff15b0f3 Mar 14
FAILING 1b54891c710e205666793dcf1933f67feba8fab9 Mar 2
FAILING 2142d76f917032d332645d9fb0cde3660432d522 Feb 27 <-
OK 438d30ac71eaa54a3916ad93a3b68966dc994204 Feb 27
OK f7718db94237cdef920b530a1dd8746f79aa32eb Feb 26
OK 830931dead61c78dbc5ab12cafd6a8d2e280a75d Feb 25
OK 772684e841f7deb175e73bcfe76a601c176026f7 Feb 24
OK cc5b3e9e8687da012839046e7e4eb26b3f0cbf8a Feb 14
OK 0250b696b12c1421931cbf6588253d8a2de5bde0 Jan 27

**Comment by jtpio** [comment]
Probably the `notebook` example needs to be updated to include extra logic from https://github.com/jupyterlab/jupyterlab/pull/13639.

For reference Notebook 7 handled this change in https://github.com/jupyter/notebook/pull/6757. But it's using the JupyterLab extensions and plugins directly which is not the case with the `notebook` example, so not sure it will be helpful.

**Comment by echarles** [comment]
> For reference Notebook 7 handled this change in https://github.com/jupyter/notebook/pull/6757. But it's using the JupyterLab extensions and plugins directly which is not the case with the notebook example, so not sure it will be helpful.

Thx for the link. I have scanned that PR but as you said, Notebook 7 is using directly the JupyterLab extensions,  so has not been impacted by those the CodeMirror Token changes directly. The only change from that PR is linked to the menu bar.

I have already looked closely to the logic changes introduced by the CodeMirror Token PR and its feels to me like the impact on the examples has not been identified. While debugging, the model is not updated (the source attribute is empty when you run the cell with content), which makes me think there miss something on the codemirror <-> model binding side.
